### PR TITLE
Move memory_layout.rs (and deps) to caliptra_drivers

### PR DESCRIPTION
**Pull request by korran**
This is necessary to allow the drivers to reference memory-layout
constants without introducing a circular dependency.

Also, take advantage of this change to remove a copy-pasted constant from report_boot_status().

Also, delete the rom printing infrastructure, and use the one in caliptra-drivers instead.

Also, don't attempt to print the FHT from the ROM, as it will actually do something now (the rom enables caliptra-drivers' emu feature, unlike caliptra-common).

### CODE REVIEW

**Review by mhatrevi — APPROVED** [review]

File: drivers/src/printer.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ impl uWrite for MutablePrinter {
     #[inline(never)]
     fn write_str(&mut self, _str: &str) -> Result<(), Self::Error> {
         #[cfg(feature = "emu")]
-        caliptra_drivers::Uart::default().write(_str);
```
Comment (rusty1968) [inline drivers/src/printer.rs:28]: nit : Instead of MutablePrinter, let's call it just Printer.

Comment (korran) [reply drivers/src/printer.rs:28]: That's the name it had before; I didn't change it.

**Review by rusty1968 — APPROVED** [review]

File: drivers/src/printer.rs (hunk 1 above)
Comment (rusty1968) [reply drivers/src/printer.rs:28]: I know. I regret naming it like that.

**Review by mhatrevi — APPROVED** [review]

**Review by jhand2 — APPROVED** [review]

File: drivers/src/printer.rs (hunk 1 above)
Comment (korran) [reply drivers/src/printer.rs:28]: Renamed.

**Review by jhand2 — APPROVED** [review]

**Review by rusty1968 — APPROVED** [review]

**Pull request merged**